File: app/src/main/java/com/team2052/frckrawler/activities/TeamInfoActivity.java
package com.team2052.frckrawler.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.team2052.frckrawler.R;
import com.team2052.frckrawler.adapters.tab.ViewTeamPagerAdapter;
import com.team2052.frckrawler.db.Team;

/**
 * @author Adam
 */
public class TeamInfoActivity extends DatabaseActivity {
    ViewPager viewPager;
    TabLayout tabLayout;

    public static Intent newInstance(Context context, Team team) {
        Intent intent = new Intent(context, TeamInfoActivity.class);
        intent.putExtra(PARENT_ID, team.getNumber());
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_tab);
        long team_id = getIntent().getLongExtra(PARENT_ID, 0);

        viewPager = (ViewPager) findViewById(R.id.view_pager);
        tabLayout = (TabLayout) findViewById(R.id.tab_layout);
        viewPager.setAdapter(new ViewTeamPagerAdapter(this, getSupportFragmentManager(), team_id));
        tabLayout.setupWithViewPager(viewPager);

        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        setActionBarSubtitle(String.valueOf(team_id));
    }

    @Override
    public void inject() {
        getComponent().inject(this);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/team2052/frckrawler/activities/TeamInfoActivity.kt
package com.team2052.frckrawler.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.team2052.frckrawler.R
import com.team2052.frckrawler.adapters.tab.TeamInfoPagerAdapter
import com.team2052.frckrawler.bindToViewPagerAndTabLayout
import kotlinx.android.synthetic.main.layout_tab.*

/**
 * @author Adam
 */
class TeamInfoActivity : DatabaseActivity() {
    companion object {

        fun newInstance(context: Context, team_number: Long): Intent {
            val intent = Intent(context, TeamInfoActivity::class.java)
            intent.putExtra(DatabaseActivity.PARENT_ID, team_number)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_tab)
        val team_id = intent.getLongExtra(DatabaseActivity.PARENT_ID, 0)

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        setActionBarSubtitle(team_id.toString())

        TeamInfoPagerAdapter(this, supportFragmentManager, team_id).bindToViewPagerAndTabLayout(view_pager, tab_layout)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun inject() {
        component.inject(this)
    }
}
